amplify/backend/function/fetchDomainBalance/src/services/tokens.js

```diff
@@ -1,38 +1,54 @@
 const { BigNumber } = require('ethers');
 const { getStartOfDayFor } = require('../utils');
 
-const getConvertedTokenAmount = (amount, networkFee, decimals, exchangeRate) => {
-    const tokenScaleFactor = BigNumber.from(10).pow(decimals);
-    const exchangeRateDecimals = 9;
-    const normalizedTokenExchangeRate = BigNumber.from(Math.round(exchangeRate * Math.pow(10, exchangeRateDecimals)));
-    const formattedNetworkFee = BigNumber.from(networkFee || 0)
+const getConvertedTokenAmount = (
+  amount,
+  networkFee,
+  decimals,
+  exchangeRate,
+) => {
+  const tokenScaleFactor = BigNumber.from(10).pow(decimals);
+  const exchangeRateDecimals = 9;
+  const normalizedTokenExchangeRate = BigNumber.from(
+    Math.round(exchangeRate * Math.pow(10, exchangeRateDecimals)),
+  );
+  const formattedNetworkFee = BigNumber.from(networkFee || 0);
 
-    return BigNumber
-        .from(amount)
-        .sub(formattedNetworkFee)
-        .div(tokenScaleFactor)
-        .mul(normalizedTokenExchangeRate)
-        .div(BigNumber.from(10).pow(exchangeRateDecimals));
-}
+  return (
+    BigNumber.from(amount)
+      // @TODO: networkFee needs to be add when calculating outgoing expenditures as it is added on top of the amount
+      // However, I am not sure if this is the case with every other action being passed here
+      // Maybe in some cases networkFee needs to be subtracted?
```

Comment:
Totally agree with this comment, though we should consider adding the `networkFee` or `-networkFee` at the actions/funds/expenditure level for better clarity 😁